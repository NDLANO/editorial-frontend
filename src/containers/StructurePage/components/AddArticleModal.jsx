import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectT } from 'ndla-i18n';
import { updateTopic } from '../../../modules/taxonomy';
import TaxonomyLightbox from './TaxonomyLightbox';
import { AsyncDropdown } from '../../../components/Dropdown';
import { searchRelatedArticles } from '../../../modules/article/articleApi';

class AddArticleModal extends Component {
  constructor() {
    super();
    this.onArticleSearch = this.onArticleSearch.bind(this);
    this.onSelect = this.onSelect.bind(this);
  }
  async onArticleSearch(input) {
    try {
      const results = await searchRelatedArticles(
        input,
        this.props.locale,
        'topic-article',
      );
      return results;
    } catch (e) {
      console.log(e);
    }
  }

  async onSelect(article) {
    try {
      const { currentTopic, refreshTopics, toggleAddModal } = this.props;
      const ok = await updateTopic({
        id: currentTopic.id,
        name: currentTopic.name,
        contentUri: `urn:article:${article.id}`,
      });
      if (ok) {
        refreshTopics();
        toggleAddModal();
      }
    } catch (e) {
      console.log(e);
    }
  }

  render() {
    const { t, toggleAddModal } = this.props;
    return (
      <TaxonomyLightbox
        title={t('taxonomy.searchArticle')}
        onClose={toggleAddModal}>
        <AsyncDropdown
          valueField="id"
          name="resourceSearch"
          textField="title.title"
          placeholder={t('form.content.relatedArticle.placeholder')}
          label="label"
          apiAction={this.onArticleSearch}
          messages={{
            emptyFilter: '',
            emptyList: t('taxonomy.noResources'),
          }}
          onChange={this.onSelect}
          alwaysOpen
        />
      </TaxonomyLightbox>
    );
  }
}

AddArticleModal.propTypes = {
  locale: PropTypes.string,
  toggleAddModal: PropTypes.func,
};

export default injectT(AddArticleModal);
