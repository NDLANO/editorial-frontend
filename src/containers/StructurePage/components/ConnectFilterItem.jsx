/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import BEMHelper from 'react-bem-helper';
import ToggleSwitch from './ToggleSwitch';
import {
  RESOURCE_FILTER_CORE,
  RESOURCE_FILTER_SUPPLEMENTARY,
} from '../../../constants';

const classes = new BEMHelper({
  name: 'connectFilter',
  prefix: 'c-',
});

const ConnectFilterItem = ({ id, name, inputValues, onChange }) => {
  const relevance =
    inputValues.relevance === undefined
      ? true
      : inputValues.relevance === RESOURCE_FILTER_CORE;
  return (
    <div {...classes('')}>
      <label {...classes('item')}>
        <input
          {...classes('checkbox')}
          type="checkbox"
          name={`${id}-active`}
          checked={inputValues.active || false}
          onChange={() => onChange({ active: !inputValues.active })}
        />
        {name}
      </label>
      <ToggleSwitch
        onClick={() =>
          onChange({
            relevance: relevance
              ? RESOURCE_FILTER_SUPPLEMENTARY
              : RESOURCE_FILTER_CORE,
          })
        }
        on={relevance}
        testId={`${id}-relevance`}
      />
    </div>
  );
};

ConnectFilterItem.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  active: PropTypes.bool,
  relevanceId: PropTypes.string,
  inputValues: PropTypes.shape({
    relevance: PropTypes.string,
    active: PropTypes.bool,
  }),
  onChange: PropTypes.func,
};

export default ConnectFilterItem;
