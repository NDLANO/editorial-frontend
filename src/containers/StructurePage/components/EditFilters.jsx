/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'ndla-ui';
import { injectT } from 'ndla-i18n';
import { Plus } from 'ndla-icons/action';
import handleError from '../../../util/handleError';
import InlineEditField from './InlineEditField';
import {
  createSubjectFilter,
  editSubjectFilter,
  deleteFilter,
} from '../../../modules/taxonomy';
import WarningModal from '../../../components/WarningModal';
import EditFilterList from './EditFilterList';

class EditFilters extends React.Component {
  constructor() {
    super();
    this.state = {
      editMode: '',
    };

    this.onCancel = this.onCancel.bind(this);
    this.addFilter = this.addFilter.bind(this);
    this.showDeleteWarning = this.showDeleteWarning.bind(this);
    this.deleteFilter = this.deleteFilter.bind(this);
    this.editFilter = this.editFilter.bind(this);
  }

  onCancel() {
    this.setState({ showDelete: '' });
  }

  async addFilter(name) {
    try {
      await createSubjectFilter(this.props.id, name);
      this.props.getFilters();
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
    }
  }

  async editFilter(id, name) {
    try {
      await editSubjectFilter(id, this.props.id, name);
      this.props.getFilters();
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
    }
  }

  showDeleteWarning(filterId) {
    this.setState({ showDelete: filterId });
  }

  async deleteFilter() {
    try {
      await deleteFilter(this.state.showDelete);
      this.props.getFilters();
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
    }
    this.setState({ showDelete: false });
  }

  render() {
    const { classes, t, filters } = this.props;
    const { editMode, showDelete } = this.state;

    return (
      <div {...classes('editFilters')} data-testid="editFilterBox">
        <EditFilterList
          filters={filters}
          editMode={editMode}
          classes={classes}
          setEditState={name => this.setState({ editMode: name })}
          showDeleteWarning={this.showDeleteWarning}
          editFilter={this.editFilter}
        />
        {editMode === 'addFilter' ? (
          <InlineEditField
            classes={classes}
            currentVal=""
            messages={{ errorMessage: t('taxonomy.errorMessage') }}
            dataTestid="addFilterInput"
            onClose={() => this.setState({ editMode: '' })}
            onSubmit={this.addFilter}
          />
        ) : (
          <Button
            stripped
            {...classes('addFilter')}
            data-testid="addFilterButton"
            onClick={() => this.setState({ editMode: 'addFilter' })}>
            <Plus />
            {t('taxonomy.addFilter')}
          </Button>
        )}
        <div {...classes('errorMessage')}>{this.state.error}</div>
        {showDelete && (
          <WarningModal
            text={t('taxonomy.confirmDelete')}
            firstAction={{ text: t('form.abort'), action: this.onCancel }}
            secondAction={{
              text: t('warningModal.delete'),
              action: this.deleteFilter,
            }}
            onCancel={this.onCancel}
          />
        )}
      </div>
    );
  }
}

EditFilters.propTypes = {
  id: PropTypes.string,
  t: PropTypes.func,
  classes: PropTypes.func,
  filters: PropTypes.arrayOf(PropTypes.object),
  getFilters: PropTypes.func,
};

export default injectT(EditFilters);
