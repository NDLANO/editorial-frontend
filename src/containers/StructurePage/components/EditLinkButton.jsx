/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'ndla-ui';
import { injectT } from 'ndla-i18n';
import { Link as LinkIcon } from 'ndla-icons/editor';
import BEMHelper from 'react-bem-helper';
import { Cross, Pencil } from 'ndla-icons/action';
import Overlay from '../../../components/Overlay';
import RoundIcon from '../../../components/RoundIcon';
import WarningModal from '../../../components/WarningModal';

const classes = new BEMHelper({
  name: 'settingsMenu',
  prefix: 'c-',
});

class EditLinkButton extends Component {
  constructor() {
    super();
    this.state = {
      open: false,
    };
  }

  render() {
    const { refFunc, id, t, setPrimary } = this.props;
    return (
      <React.Fragment>
        {this.state.open && (
          <Overlay onExit={() => this.setState({ open: false })} />
        )}
        {this.state.showWarning && (
          <WarningModal
            text={t('taxonomy.confirmSetPrimary')}
            onCancel={() => this.setState({ showWarning: false })}
            onContinue={() => setPrimary(id)}
          />
        )}
        <div
          style={{ display: 'none' }}
          ref={el => (el ? refFunc(el, `linkButton-${id}`) : undefined)}>
          <Button stripped onClick={() => this.setState({ open: true })}>
            <RoundIcon icon={<LinkIcon />} />
          </Button>
          {this.state.open && (
            <React.Fragment>
              <div {...classes('openMenu')}>
                <div className="header">
                  <RoundIcon icon={<LinkIcon />} open />
                  <span>{t(`taxonomy.linkSettings`)}</span>
                  <Button
                    stripped
                    {...classes('closeButton')}
                    onClick={() => this.setState({ open: false })}>
                    <Cross />
                  </Button>
                </div>
                <Button
                  stripped
                  {...classes('menuItem')}
                  onClick={() => this.setState({ showWarning: true })}>
                  <RoundIcon small icon={<Pencil />} />
                  {t('taxonomy.setPrimary')}
                </Button>
              </div>
            </React.Fragment>
          )}
        </div>
      </React.Fragment>
    );
  }
}

EditLinkButton.propTypes = {
  refFunc: PropTypes.func,
  id: PropTypes.string,
  setPrimary: PropTypes.func,
};

export default injectT(EditLinkButton);
