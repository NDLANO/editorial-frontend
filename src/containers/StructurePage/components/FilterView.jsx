import React from 'react';
import PropTypes from 'prop-types';
import BEMHelper from 'react-bem-helper';

const classes = new BEMHelper({
  name: 'filter-view',
  prefix: 'c-',
});

const Filter = ({ name, id, toggleFilter, active }) => (
  <label {...classes('item')}>
    <input
      {...classes('checkbox')}
      checked={!!active}
      onChange={() => toggleFilter(id)}
      type="checkbox"
    />
    <span>{name}</span>
  </label>
);

Filter.propTypes = {
  name: PropTypes.string,
  toggleFilter: PropTypes.func,
  id: PropTypes.string,
  active: PropTypes.string,
};

const FilterView = ({ subjectFilters, activeFilters, toggleFilter }) => (
  <div {...classes('wrapper')}>
    {subjectFilters.map(filter => (
      <Filter
        {...filter}
        key={filter.id}
        toggleFilter={toggleFilter}
        active={activeFilters.find(filterId => filterId === filter.id)}
      />
    ))}
  </div>
);

FilterView.propTypes = {
  subjectFilters: PropTypes.arrayOf(PropTypes.object),
  activeFilters: PropTypes.arrayOf(PropTypes.string),
  toggleFilter: PropTypes.func,
};

export default FilterView;
