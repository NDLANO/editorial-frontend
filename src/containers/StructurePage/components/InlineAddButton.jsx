/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'ndla-ui';
import { Done } from 'ndla-icons/editor';
import BEMHelper from 'react-bem-helper';

const classes = new BEMHelper({
  name: 'addButton',
  prefix: 'c-',
});

class InlineAddButton extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      editMode: false,
      inputValue: '',
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  handleClick(e) {
    e.stopPropagation();
    this.props.action(this.state.inputValue);
    this.setState({ editMode: false });
  }

  handleInputChange(e) {
    e.stopPropagation();
    this.setState({ inputValue: e.target.value });
  }

  handleKeyPress(e) {
    if (e.key === 'Escape') {
      this.setState({ editMode: false });
    }
    if (e.key === 'Enter') {
      this.props.action(this.state.inputValue);
      this.setState({ editMode: false });
    }
  }

  render() {
    const { title } = this.props;

    return this.state.editMode ? (
      <div {...classes('editMode')}>
        <input
          type="text"
          data-testid="addSubjectInputField"
          value={this.state.inputValue}
          onChange={this.handleInputChange}
          onKeyDown={this.handleKeyPress}
        />
        <Button stripped onClick={this.handleClick}>
          <Done />
        </Button>
      </div>
    ) : (
      <Button
        stripped
        data-testid="AddSubjectButton"
        onClick={() => {
          this.setState({ editMode: true });
        }}
        {...classes('')}>
        {title}
      </Button>
    );
  }
}

InlineAddButton.propTypes = {
  title: PropTypes.string.isRequired,
  action: PropTypes.func,
};

export default InlineAddButton;
