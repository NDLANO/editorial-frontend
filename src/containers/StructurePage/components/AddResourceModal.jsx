import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectT } from 'ndla-i18n';
import TaxonomyLightbox from './TaxonomyLightbox';
import { AsyncDropdown } from '../../../components/Dropdown';
import { groupSearch } from '../../../modules/search/searchApi';
import {
  createTopicResource,
  fetchResource,
  fetchResourceResourceType,
} from '../../../modules/taxonomy';
import { getResourceIdFromPath } from '../../../util/routeHelpers';

import { getArticle } from '../../../modules/article/articleApi';
import ArticlePreview from './ArticlePreview';

class AddResourceModal extends Component {
  constructor() {
    super();
    this.state = {
      selected: {},
      article: {},
      pastedUrl: '',
    };
    this.onSelect = this.onSelect.bind(this);
    this.addSelected = this.addSelected.bind(this);
    this.onPaste = this.onPaste.bind(this);
    this.articleToState = this.articleToState.bind(this);
  }

  onSelect(selected) {
    if (selected) {
      const articleId = selected.url.split('/').pop();
      this.articleToState(articleId);
      this.setState({ selected });
    } else {
      this.setState({ selected: {}, article: {} });
    }
  }

  async onPaste(e) {
    const val = e.target.value;
    const resourceId = getResourceIdFromPath(val);

    if (resourceId) {
      try {
        const [resource, resourceType] = await Promise.all([
          fetchResource(resourceId),
          fetchResourceResourceType(resourceId),
        ]);
        this.articleToState(resource.contentUri.split(':').pop());
        const type = resourceType.length > 0 && resourceType[0].id;
        const error =
          type === this.props.type
            ? ''
            : `${this.props.t('taxonomy.wrongType')} ${type}`;
        this.setState({ selected: { id: val }, pastedUrl: val, error });
      } catch (error) {
        console.log(error);
      }
    } else if (!val) {
      this.setState({ error: '', pastedUrl: val });
    } else {
      this.setState({ error: 'Invalid url', pastedUrl: val });
    }
  }

  async articleToState(articleId) {
    const {
      id,
      metaDescription = {},
      title = {},
      metaImage = {},
    } = await getArticle(articleId);
    this.setState({
      article: {
        id,
        metaDescription: metaDescription.metaDescription,
        title: title.title,
        imageUrl: metaImage.url,
      },
    });
  }

  async addSelected() {
    const { topicId, refreshResources, onClose } = this.props;
    const { selected } = this.state;
    if (selected.id) {
      try {
        this.setState({ loading: true });
        await createTopicResource({
          resourceId: getResourceIdFromPath(selected.id),
          topicid: `urn:${topicId}`,
        });
        refreshResources();
        this.setState({ loading: false });
        onClose();
      } catch (e) {
        console.log(e);
        this.setState({ loading: false, error: e.message });
      }
    }
  }

  render() {
    const { onClose, type, t } = this.props;
    const { selected, article, loading, pastedUrl, error } = this.state;

    return (
      <TaxonomyLightbox
        title={t('taxonomy.searchResource')}
        onSelect={this.addSelected}
        loading={loading}
        onClose={onClose}>
        <input
          type="text"
          data-testid="addResourceUrlInput"
          value={pastedUrl}
          onChange={this.onPaste}
          placeholder={t('taxonomy.urlPlaceholder')}
        />
        {error && <span className="c-errorMessage">{error}</span>}

        {!pastedUrl && (
          <React.Fragment>
            <span
              style={{
                color: 'white',
                alignSelf: 'center',
              }}>
              {t('taxonomy.or')}
            </span>
            <AsyncDropdown
              valueField="id"
              name="resourceSearch"
              textField="title"
              placeholder={t('form.content.relatedArticle.placeholder')}
              label="label"
              apiAction={async inp => {
                const res = await groupSearch(inp, type);
                const result = res.length > 0 ? res.pop().results : [];
                return result.map(curr => ({
                  ...curr,
                  id: curr.paths.pop(),
                  title: curr.title ? curr.title.title : '',
                }));
              }}
              messages={{
                emptyFilter: '',
                emptyList: t('taxonomy.noResources'),
              }}
              onChange={this.onSelect}
              alwaysOpen
            />
          </React.Fragment>
        )}
        {selected.id && article.id && <ArticlePreview article={article} />}
      </TaxonomyLightbox>
    );
  }
}

AddResourceModal.propTypes = {
  onClose: PropTypes.func,
  type: PropTypes.string,
};

export default injectT(AddResourceModal);
