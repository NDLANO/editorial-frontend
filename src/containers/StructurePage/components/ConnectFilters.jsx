/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'ndla-ui';
import { injectT } from 'ndla-i18n';
import {
  fetchTopicFilters,
  addFilterToTopic,
  updateTopicFilter,
  deleteTopicFilter,
} from '../../../modules/taxonomy';
import ConnectFilterItem from './ConnectFilterItem';
import Spinner from '../../../components/Spinner';
import Overlay from '../../../components/Overlay';
import { RESOURCE_FILTER_CORE } from '../../../constants';

class ConnectFilters extends Component {
  constructor() {
    super();
    this.state = {
      topicFilters: [],
      inputs: {},
      loading: false,
    };
    this.getFilters = this.getFilters.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentDidMount() {
    this.getFilters();
  }

  async onSubmit() {
    const { subjectFilters, id } = this.props;
    const { topicFilters, inputs } = this.state;
    this.setState({ loading: true });
    await Promise.all([
      ...subjectFilters.filter(filter => !!inputs[filter.id]).map(filter => {
        const { active, relevance } = inputs[filter.id];
        const currentFilter = topicFilters.find(it => it.id === filter.id);
        if (active && !currentFilter) {
          // add filter to topic
          return addFilterToTopic({
            filterId: filter.id,
            topicId: id,
            relevanceId: relevance || RESOURCE_FILTER_CORE,
          });
        } else if (active && relevance !== currentFilter.relevanceId) {
          // update topic-filter with relevance
          return updateTopicFilter({
            connectionId: currentFilter.connectionId,
            relevanceId: relevance,
          });
        } else if (!active && currentFilter) {
          // delete topic-filter
          return deleteTopicFilter({
            connectionId: currentFilter.connectionId,
          });
        }
        return undefined;
      }),
    ]);
    const refreshedTopics = await fetchTopicFilters(this.props.id);
    const inputVals = refreshedTopics.reduce((acc, curr) => {
      acc[curr.id] = { active: true, relevance: curr.relevanceId };
      return acc;
    }, {});
    this.setState({
      loading: false,
      topicFilters: refreshedTopics,
      inputs: inputVals,
    });
  }

  async getFilters() {
    const topicFilters = await fetchTopicFilters(this.props.id);
    const inputs = topicFilters.reduce((acc, curr) => {
      acc[curr.id] = { active: true, relevance: curr.relevanceId };
      return acc;
    }, {});
    this.setState({ topicFilters, inputs, error: '' });
  }

  render() {
    const { subjectFilters, classes, t } = this.props;
    const { inputs, error, loading } = this.state;
    return (
      <form onSubmit={this.onSubmit} {...classes('editFilters')}>
        {loading && <Spinner cssModifier="absolute" />}
        {loading && (
          <Overlay cssModifiers={['absolute', 'white-opacity', 'zIndex']} />
        )}
        {subjectFilters.map(filter => (
          <ConnectFilterItem
            {...filter}
            inputValues={inputs[filter.id] || {}}
            onChange={val =>
              this.setState({
                inputs: {
                  ...inputs,
                  [filter.id]: {
                    ...inputs[filter.id],
                    ...val,
                  },
                },
              })
            }
            key={filter.id}
          />
        ))}
        <Button
          stripped
          {...classes('borderButton')}
          data-testid="submitConnectFilters"
          onClick={this.onSubmit}>
          {t('form.save')}
        </Button>
        {error && <div {...classes('errorMessage')}>{error}</div>}
      </form>
    );
  }
}

ConnectFilters.propTypes = {
  classes: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  subjectFilters: PropTypes.array,
};

export default injectT(ConnectFilters);
