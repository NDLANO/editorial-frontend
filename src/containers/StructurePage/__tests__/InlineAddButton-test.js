/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import { InlineAddButton } from '../../../components/InlineAddButton';

afterEach(cleanup);

it('Goes to edit mode, handles input and calls action prop', async () => {
  const actionFunc = jest.fn();
  const { getByTestId, container } = render(
    <InlineAddButton title="Test" action={actionFunc} />,
  );
  expect(container.firstChild).toMatchSnapshot();

  fireEvent.click(getByTestId('AddSubjectButton'));
  const input = getByTestId('addSubjectInputField');
  input.value = 'Elefant';
  fireEvent.change(input);
  expect(container.firstChild).toMatchSnapshot();

  fireEvent.keyDown(input, { key: 'Enter', keyCode: 13, which: 13 });
  expect(container.firstChild).toMatchSnapshot();
  expect(actionFunc).toHaveBeenCalledTimes(1);
});
