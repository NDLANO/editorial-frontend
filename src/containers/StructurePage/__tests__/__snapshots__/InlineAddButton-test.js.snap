// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goes to edit mode, handles input and calls action prop 1`] = `
<button
  class="c-button c-button--stripped c-addButton"
  data-testid="AddSubjectButton"
  type="button"
>
  <svg
    class="c-icon"
    data-license="Apache License 2.0"
    data-source="Material Design"
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align: middle;"
    viewBox="0 0 24 24"
    width="1em"
  >
    <title>
      Plus
    </title>
    <g>
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </g>
  </svg>
  Test
</button>
`;

exports[`Goes to edit mode, handles input and calls action prop 2`] = `
<div
  class="c-addButton__editMode"
>
  <input
    data-testid="addSubjectInputField"
    type="text"
    value="Elefant"
  />
  <button
    class="c-button c-button--stripped"
    type="button"
  >
    <svg
      class="c-icon"
      data-license="Apache License 2.0"
      data-source="Material Design"
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style="vertical-align: middle;"
      viewBox="0 0 24 24"
      width="1em"
    >
      <title>
        Done
      </title>
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`Goes to edit mode, handles input and calls action prop 3`] = `
<div
  class="c-addButton__editMode"
>
  <input
    data-testid="addSubjectInputField"
    type="text"
    value="Elefant"
  />
  <button
    class="c-button c-button--stripped"
    disabled=""
    type="button"
  >
    <div
      class="c-spinner c-spinner--small"
    />
  </button>
</div>
`;
